publication-doi-commons/src/main/java/no/unit/nva/publication/doi/PublicationMapper.java

```diff
@@ -94,10 +98,28 @@ public Publication fromDynamodbStreamRecordDao(DynamodbStreamRecordImageDao dao)
             .withType(extractPublicationInstanceType(dao))
             .withPublicationDate(new PublicationDate(dao.getPublicationReleaseDate()))
             .withDoi(extractDoiUrl(dao))
+            .withDoiRequest(extractDoiRequest(dao))
+            .withModifiedDate(Instant.parse(dao.getModifiedDate()))
+            .withStatus(PublicationStatus.valueOf(dao.getStatus().toUpperCase(Locale.getDefault())))
             .withContributor(ContributorMapper.fromIdentityDaos(dao.getContributorIdentities()))
             .build();
     }
 
+    private DoiRequest extractDoiRequest(DynamodbStreamRecordImageDao dao) {
+        var jsonPointers = new DynamodbStreamRecordJsonPointers(DynamodbImageType.NONE);
+        JsonNode doiRequest = dao.getDoiRequest();
+        return new DoiRequest.Builder()
+            .withStatus(DoiRequestStatus.valueOf(doiRequest.at(
```

Comment:
refactor the arg of value of into a method. Parsing of string should be inside the enum of DoiRequestStatus and should be ignoring case.

Reply:
updated